src/components/notebooks/modal.tsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from 'react';
+import { XMarkIcon } from '@heroicons/react/24/solid';
+
+export default function Modal({
+  buttonTitle,
+  modalTitle,
+  buttonStyle,
+  children,
+}: {
+  buttonTitle: string;
+  modalTitle: string;
+  buttonStyle: string;
+  children: React.ReactNode;
+}) {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const toggleModal = () => {
+    setIsModalOpen(!isModalOpen);
+  };
+
+  return (
+    <>
+      <button
+        type="button"
+        className={buttonStyle}
+        onClick={toggleModal}
+      >
+        {buttonTitle}
+      </button>
+
+      {isModalOpen && (
+        <div
+          id="authentication-modal"
```

Comment:
`id="authentication-modal"`? In relation to the other comment I made about it, this should probably be one of the props that allow us to toggle the modal.